data/glsk/glsk-document-api/src/main/java/com/farao_community/farao/data/glsk/api/util/converters/GlskPointScalableConverter.java

```diff
@@ -123,25 +123,43 @@ private static Scalable convertMeritOrder(Network network, AbstractGlskPoint gls
-                    return (Scalable) Scalable.onGenerator(generatorId, incomingMinP, incomingMaxP);
+                    return getGeneratorScalableWithLimits(network, generatorRegisteredResource);
                 }).toArray(Scalable[]::new));
 
         Scalable downScalable = Scalable.stack(glskPoint.getGlskShiftKeys().stream()
                 .filter(abstractGlskShiftKey -> abstractGlskShiftKey.getMeritOrderPosition() < 0)
                 .sorted(Comparator.comparingInt(AbstractGlskShiftKey::getMeritOrderPosition).reversed())
                 .map(glskShiftKey -> {
                     AbstractGlskRegisteredResource generatorRegisteredResource = Objects.requireNonNull(glskShiftKey.getRegisteredResourceArrayList()).get(0);
-                    String generatorId = generatorRegisteredResource.getGeneratorId();
-                    double incomingMaxP = generatorRegisteredResource.getMaximumCapacity().orElse(Double.MAX_VALUE);
-                    double incomingMinP = generatorRegisteredResource.getMinimumCapacity().orElse(-Double.MAX_VALUE);
-                    return (Scalable) Scalable.onGenerator(generatorId, incomingMinP, incomingMaxP);
+                    return getGeneratorScalableWithLimits(network, generatorRegisteredResource);
                 }).toArray(Scalable[]::new));
         return Scalable.upDown(upScalable, downScalable);
     }
 
+    private static Scalable getGeneratorScalableWithLimits(Network network, AbstractGlskRegisteredResource generatorRegisteredResource) {
+        String generatorId = generatorRegisteredResource.getGeneratorId();
+        double incomingMaxP = generatorRegisteredResource.getMaximumCapacity().orElse(Double.MAX_VALUE);
+        double incomingMinP = generatorRegisteredResource.getMinimumCapacity().orElse(-Double.MAX_VALUE);
+        // Fixes some inconsistencies between GLSK and network that may raise an exception in
+        // PowSyBl when actually scaling the network.
+        // TODO: Solve this issue in PowSyBl framework.
+        Generator generator = network.getGenerator(generatorId);
+        if (generator != null) {
+            double generatorTargetP = generator.getTargetP();
+            if (!Double.isNaN(incomingMaxP) && incomingMaxP < generatorTargetP) {
+                LOGGER.warn("Generator '{}' has initial target P that is above GLSK target P. Extending GLSK max P from {} to {}.", generatorId, incomingMaxP, generatorTargetP);
+                incomingMaxP = generatorTargetP;
+            }
+            if (!Double.isNaN(incomingMinP) && incomingMinP > generatorTargetP) {
+                LOGGER.warn("Generator '{}' has initial target P that is above GLSK target P. Extending GLSK min P from {} to {}.", generatorId, incomingMinP, generatorTargetP);
+                incomingMinP = generatorTargetP;
+            }
+            incomingMaxP = Math.max(incomingMaxP, generator.getTargetP());
+            incomingMinP = Math.min(incomingMinP, generator.getTargetP());
```

Comment:
Isn't it redundant with what is done previously ?

Reply:
No, previously, the check that is done is "if GLSK does not provide max, we use Double.max" and "if  GLSK does not provide min, we use -Double.max"

Reply:
I understood the remark, fixed